src/osdc/Striper.cc

```diff
@@ -390,12 +390,14 @@ void Striper::StripedReadResult::add_partial_result(
 {
   ldout(cct, 10) << "add_partial_result(" << this << ") " << bl.length()
 		 << " to " << buffer_extents << dendl;
+  uint64_t off = 0;
   for (auto p = buffer_extents.cbegin(); p != buffer_extents.cend(); ++p) {
-    pair<bufferlist, uint64_t>& r = partial[p->first];
+    pair<bufferlist, uint64_t>& r = partial[off];
```

Comment:
This change is wrong -- if something is broken, it's implying an issue in librbd/RWL

Reply:
Yes.